auth/auth/driver/driver.py

```diff
@@ -481,21 +398,14 @@ async def _create_user(app, user, skip_trial_bp, cleanup):
         updates['hail_credentials_secret_name'] = hail_credentials_secret_name
 
     namespace_name = user['namespace_name']
-    if namespace_name is None and user['is_developer'] == 1:
+    # auth services in test namespaces cannot/should not be creating and deleting namespaces
```

Comment:
I think I would want a check here if it is a dev deployment that the expected secret given by `hail_credentials_secret_name` exists in that namespace using the k8s client.